packages/feature-detector/scripts/patch-type-definition.ts

```diff
@@ -0,0 +1,106 @@
+// assumes being ran in packages/feature-detector
+
+import { get } from 'node:https'
+import { writeFile } from 'node:fs/promises'
+import { join } from 'node:path'
+
+const repositoryBaseUrl =
+  'https://raw.githubusercontent.com/microsoft/TypeScript/main/lib'
+const destinationFile = join('src', 'utils/type-definition.txt')
+
+async function writeTypeDefinition() {
+  const aggregatedContent = await fetchTypeDefinitions([
+    'lib.es2022.d.ts',
+    'lib.dom.d.ts',
+  ])
+  await writeFile(destinationFile, patchContent(aggregatedContent).join('\n'))
+}
+
+async function fetchTypeDefinitions(files: string[]) {
+  const result = []
+  for (const file of files) {
+    result.push(...(await fetchTypeDefinition(file)))
+  }
+  return result
+}
+
+async function fetchTypeDefinition(file: string): Promise<string[]> {
+  // I wish we could use node@18's fetch...
+  const content = await new Promise<string>((resolve, reject) =>
+    get(`${repositoryBaseUrl}/${file}`, (response) => {
+      let content = ''
+      response.setEncoding('utf8')
+      response.on('data', (chunk: string) => (content += chunk))
+      response.once('end', () => resolve(content))
+      response.once('error', reject)
+    })
+  )
+  const result = []
+  for (const line of content.split('\n')) {
+    const reference = hasReference(line)
+    if (reference) {
+      result.push(...(await fetchTypeDefinition(`lib.${reference}.d.ts`)))
+    } else {
+      result.push(line)
+    }
+  }
+  return result
+}
+
+function hasReference(line: string) {
+  const match = line.match(/^\/\/\/\s*<reference\s+lib\s*=\s*"(.+)"\s*\/>/)
+  return match ? match[1] : null
+}
+
+function patchContent(lines: string[]) {
+  const result = []
+  let inComment = false
+  for (const line of lines) {
+    if (inComment) {
+      if (isMultiCommentEnd(line)) {
+        inComment = false
+      }
+    } else {
+      if (isMultiCommentStart(line)) {
+        inComment = true
+      } else if (!isSingleComment(line) && !isBlankLine(line)) {
+        result.push(...applyPatch(line))
+      }
+    }
+  }
+  return result
+}
+
+function applyPatch(line: string) {
+  if (line === 'declare var Response: {') {
```

Comment:
> Because `Response.json()` static method is not supported in Typescript default types ([lib.dom.d.ts](https://github.com/microsoft/TypeScript/blob/4b52d3a61a05a7c2b3d03a8e1c801560232f3194/lib/lib.dom.d.ts#L11782-L11787)), I had to overload default types with an augmented definition.